src/daemon/rpmostreed-os.c

```diff
@@ -77,6 +81,101 @@ sysroot_changed (RpmostreedSysroot *sysroot,
     g_warning ("%s", local_error->message);
 }
 
+static gboolean
+os_authorize_method (GDBusInterfaceSkeleton *interface,
+                     GDBusMethodInvocation  *invocation)
+{
+  RpmostreedOS *self = RPMOSTREED_OS (interface);
+  const gchar *method_name = g_dbus_method_invocation_get_method_name (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  GVariant *parameters = g_dbus_method_invocation_get_parameters (invocation);
+  const gchar *action = NULL;
+  gboolean authorized = FALSE;
+
+  if (g_strcmp0 (method_name, "GetDeploymentsRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedDeployRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadDeployRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedUpdateRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadUpdateRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "GetCachedRebaseRpmDiff") == 0 ||
+      g_strcmp0 (method_name, "DownloadRebaseRpmDiff") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.repo-refresh";
+    }
+  else if (g_strcmp0 (method_name, "Deploy") == 0 ||
+           g_strcmp0 (method_name, "Rebase") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.rebase";
+    }
+  else if (g_strcmp0 (method_name, "Upgrade") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.upgrade";
+    }
+  else if (g_strcmp0 (method_name, "SetInitramfsState") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.set-initramfs-state";
+    }
+  else if (g_strcmp0 (method_name, "Cleanup") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.cleanup";
+    }
+  else if (g_strcmp0 (method_name, "Rollback") == 0 ||
+           g_strcmp0 (method_name, "ClearRollbackTarget") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.rollback";
+    }
+  else if (g_strcmp0 (method_name, "PkgChange") == 0)
+    {
+      action = "org.projectatomic.rpmostree1.package-install-uninstall";
+    }
+  else if (g_strcmp0 (method_name, "UpdateDeployment") == 0)
```

Comment:
Yeah, this is going to make this especially tricky now that every method can end up using `UpdateDeployment`!